lib/index.js

```diff
@@ -437,13 +438,45 @@ FSTree.prototype.mkdirSync = function(relativePath) {
   }
 
   let normalizedPath = this._normalizePath(relativePath);
+
   fs.mkdirSync(`${this.root}${normalizedPath}`);
   entry = new Entry(normalizedPath, 0, Date.now(), Entry.DIRECTORY_MODE, null);
 
   this._track('mkdir', entry);
   this._insertAt(result, entry);
 };
 
+
+FSTree.prototype.mkdirpSync = function(relativePath) {
+  this._throwIfStopped('mkdirp');
+  this._ensureEntriesPopulated();
+
+  let result = this.findByRelativePath(relativePath);
+  let entry = result.entry;
+  if (entry) {
+    logger.info('mkdirpSync %s noop, directory exists', relativePath);
+    return;
+  }
+
+  let paths = relativePath.split("/");
+  let subsetPaths = [];
+
+  //For path a/b/c/d, split it into a, a/b,  a/b/c, a/b/c/d and for each path check if there is an entry for that relative path
+  // if there is no entry call mkdirSync to create.
+  for (let i = 0; i < paths.length; i ++ ) {
+    if (i != 0) {
+      subsetPaths[i] = subsetPaths[i-1] + "/" + paths[i];
+    } else {
+      subsetPaths[i] = paths[i];
+    }
+
+    let result = this.findByRelativePath(subsetPaths[i]);
```

Comment:
This algorithm is O(N^2) but we can do it in O(N) because `entries` is sorted.

I don't think we need to fix it for now; it's just something to note